pubnub/pubnub_core.py

```diff
@@ -98,8 +101,12 @@ class PubNubCore:
 
     _subscription_registry: PNSubscriptionRegistry
 
-    def __init__(self, config):
-        self.config = config
+    def __init__(self, config: PNConfiguration):
+        if not config.disable_config_locking:
+            config.lock()
```

Comment:
if, as a user, I wanted to reuse most of the settings of the config to create a new (second) PN instance, is there a way I can do it now? probably not, since the original config will be locked (both outside and the inner copy in PN)?

Can you add a method to config that, even with a locked config, lets me clone+modify an existing config? or is it already possible with some Python magic, or similar to the deepcopy function that you're using here

I'd like to do something like (pseudocode):

```
val config = PNConfiguration(blah, blah, blah)
val pubnub1 = PubNub(config)

val config2 = config.copy(uuid = blah2)
val pubnub2 = PubNub(config2)
```